mypy/test/data.py

```diff
@@ -250,6 +250,7 @@ def __init__(self,
         self.line = line
         self.old_cwd: Optional[str] = None
         self.tmpdir: Optional[tempfile.TemporaryDirectory[str]] = None
+        os.environ["__MYPY_UNDER_TEST__"] = "1"
```

Comment:
what exactly is this for? it needs to behave differently when being tested? that sounds sus

Reply:
It sets `mypy.options._based` to be `False` globally so that the output can still be old compatible for tests. I think it's okay to keep the old test compatibility to avoid excess work from upstream changes.

Reply:
does that mean the based version doesnt get tested at all?

what if there was a seperately maintained set of outputs from tests run in based mode? or maybe thats too much effort as the outputs are already a nightmare to maintain

Reply:
Based test cases setill get tested under based mode